OmiseSwift/DateComponentsConverter.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+
+public class DateComponentsConverter: Converter {
+    public typealias Target = NSDateComponents
+    
+    static func makeScanner(forString string: String) -> NSScanner {
+        let scanner = NSScanner(string: string)
+        let validCharacterSet = NSCharacterSet(charactersInString: "0123456789-")
+        scanner.charactersToBeSkipped = validCharacterSet.invertedSet
+        return scanner
+    }
+    
+    public static func convertFromAttribute(value: NSObject?) -> Target? {
+        guard let s = value as? String else { return nil }
+        let scanner = makeScanner(forString: s)
+        let year: Int
+        let month: Int
+        let day: Int
+        
+        var firstInt: Int = 0
+        var secondInt: Int = 0
+        var lastInt: Int = 0
+        guard scanner.scanInteger(&firstInt) &&
+            scanner.scanString("-", intoString: nil) &&
+            scanner.scanInteger(&secondInt) &&
+            scanner.scanString("-", intoString: nil) &&
+            scanner.scanInteger(&lastInt) &&
+            scanner.scanString("-", intoString: nil)
+            else {
+                return nil
+        }
+        
+        month = secondInt
+        let isYearFirst = firstInt > 12
```

Comment:
Search API Interface supports both format. Should we also support both format too?